wrappers/firstelastic.py

```diff
@@ -1,106 +1,106 @@
+        """
+        Builds a dictionary mapping of normal FRC event keys to FIRST Event IDs used in ES.
+        
+        :param year: Int year to fetch events for.
+        :return: Dictionary containing the event key to ID map.
+        """
+        event_list = [hit['_source'] for hit in self._get(self.EVENT_LIST_URL % year)['hits']['hits']]
+        event_key_map = {}
+        for event in event_list:
+            event_key = str(year) + event['event_code'].lower()
+            event_key_map[event_key] = event['id']
+
+        return event_key_map
+
+    def event_teams(self, event: str, simple: bool = False, keys: bool = False) -> Union[List[str], List[Dict]]:
+        """
+        Get list of teams at an event. 
+        NOTE: Home championships are not calculated.
+        
+        :param event: Event key to get data on.
+        :param simple: Get only vital data.
+        :param keys: Return list of team keys only rather than full data on every team.
+        :return: List of string keys or team objects.
+        """
+
+        """Map of FIRST ES field names onto TBA names"""
+        field_map = {'website': 'team_web_url',
+                     'team_number': 'team_number_yearly',
+                     'state_prov': 'team_stateprov',
+                     'rookie_year': 'team_rookieyear',
+                     'postal_code': 'team_postalcode',
+                     'nickname': 'team_nickname',
+                     'name': 'team_name_calc',
+                     'country': 'countryCode',
+                     'city': 'team_city'
+                     }
+
+        if event in self.event_key_map:
+            first_event_id = self.event_key_map[event]
+            raw_list = [hit['_source'] for hit in
+                        self._get(self.EVENT_TEAMS_URL % (first_event_id, self.current_year))['hits']['hits']]
+
+            team_list = []
+
+            for team in raw_list:
+                team_obj = {'motto': None,
+                            'location_name': None,
+                            'key': 'frc' + str(team['team_number_yearly']),
+                            'gmaps_place_id': None,
+                            'address': None,
+                            'lng': None,
+                            'lat': None}
+
+                if 'location' in team.keys():
+                    team_obj['lng'] = team['location'][0]['lon']
+                    team_obj['lat'] = team['location'][0]['lat']
+
+                for field in field_map:
+                    if field_map[field] in team.keys():
+                        team_obj[field] = team[field_map[field]]
+                team_list.append(team_obj)
+
+            if keys:
+                return [team['key'] for team in team_list]
+
+            if simple:
+                simple_fields = ['city', 'country', 'key', 'name', 'nickname', 'state_prov', 'team_number']
+                return [{key: val for key, val in team.items() if key in simple_fields} for team in team_list]
```

Comment:
on one hand, this is nifty one-liner. on the other hand, there is a lot going on here. maybe think about splitting this up?

Reply:
@PChild